### ISSUE TITLE
Steam Deck - Idling for at 1040MHz on Home screen

### ISSUE THREAD

**Issue by Kagukara** [ValveSoftware/SteamOS#1245]
#### Your system information

- Steam client version:
- SteamOS version: 3.5.7 (Build: 20231122.1)
- Opted into Steam client beta?: No
- Opted into SteamOS beta?: No
- Have you checked for updates in Settings > System?: Yes

#### Please describe your issue in as much detail as possible:
The animations don't stop on the "Home" screen, causing the GPU to stay at 1040MHz because of the forced upon "Moving between 200/1040MHz is expected." in 3.5.5. (https://github.com/ValveSoftware/SteamOS/issues/1241#issuecomment-1828474301)

#### Performance settings used:

- Performance Overlay Level: 4
- Frame limit: 60 FPS
- Disable Frame Limit: Off
- Allow Tearing: Off
- Half Rate Shading: Off
- TDP Limit: Off
- Manual GPU Clock: Off
- Scaling Mode: Auto
- Scaling Filter: Linear
- Show Perf Overlay in Steam: On

Setting the "Manual GPU Clock: Off" to "On" and setting the "GPU Clock Frequency" to "200" causes the frequency to stay at 200MHz, temperature to lower/lowers and less power is used.

This then ends up impacting games, if the game can't push the Steam Deck higher than 1040MHz on the GPU Clock Frequency then it will switch between 200MHz and 1040MHz, showing no in between clocks.

Could you not add an option in the performance settings to allow the user to set a custom/turn off "an efficient clock to get work done fast" setting?

This would solve most if not everyone's issue with the forced upon "Moving between 200/1040MHz is expected." in 3.5.5.

Steps for reproducing this issue:

1. Turn on the Steam Deck
2. Set the Performance Settings to the default
3. Move around the menus/settings/anywhere in the Steam Deck

**Comment by Kagukara** [comment]
If I stop all the animations using CSS Loader from Decky Loader, it idles at 200MHz. With the animations it idles at 1040MHz. This is with the LCD Steam Deck.

But once you start moving around the Steam Deck it shoots all the way to 1040MHz, which isn't needed. 200MHz is adequate enough.

How would I disable this, or change the frequency it goes to, without locking it in the performance settings? Locking it in the performance settings cause those to carry across to desktop mode which causes issues.

With the Steam Deck idling at 1040MHz it heats up the Steam Deck and uses more power compared to it idling at 200MHz.

**Comment by Oliver-1xok** [comment]
> But once you start moving around the Steam Deck it shoots all the way to 1040MHz,

Of course, it goes up, but mostly stays below 1040. Of course, you should also uninstall Decky for testing. Factory settings are best. But that won't even be necessary, because I think you're looking at the inadequate output of MangoHud or something else. 

Often shows 1040 Mhz. Even if you go to the home screen of a game without animation. The deck definitely uses 200 Mhz if you do not make any input. Nevertheless, MangoHud shows 1040Mhz on my deck.

Therefore, when you examine frequencies, it is very crucial how you determine them. But you don't write anything about that. 

So I don't know what to do with your report. Where exactly do you see 1040 Mhz? I assume in MangoHud, but I can only guess. Be sure to connect to your deck via SSH for testing. Go to a page without animation (for example the Portal2 home screen). Do not touch the deck afterwards. Now execute the following command on the console:

` watch -n 0.1 od -An -tu2 -j54 -N14 /sys/class/drm/card0/device/gpu_metrics`

This is what Linux/SteamOS thinks it sees for a GPU frequency. When in doubt, I prefer to rely on the operating system for the metrics. But in any case, you must specify how you determine the frequency. Otherwise it is completely undefined. This makes it difficult to say anything about it, because perhaps everyone determines the frequency differently.

**Comment by tarmo888** [comment]
> So I don't know what to do with your report. Where exactly do you see 1040 Mhz? I assume in MangoHud

Obviously they mean MangoHud, they even mention "Performance Overlay Level: 4" and "Show Perf Overlay in Steam: On".

But thanks for the console command.

**Comment by Oliver-1xok** [comment]
https://github.com/ValveSoftware/SteamOS/issues/1245#issuecomment-1834580951
> I thought writing "Performance Overlay Level: 4" in my issue was more than enough to know what I was using to get the frequency.

Yes, of course. It was my problem. 

https://github.com/ValveSoftware/SteamOS/issues/1245#issuecomment-1834580951
> Although Portal 2 still has animations

Just to avoid misunderstandings: When I say "Portal 2 home screen", I simply mean Steam's home screen of Portal 2. I do **not** start the game. I thought there were games with moving images on their home screen. But apparently I was wrong. I've just looked through my games. We can probably use any home screen from any game.  I just want to put the Deck in a state where the GPU is permanently clocked down to 200 Mhz because it doesn't have to draw anything. I think you do that via Decky.    

When the GPU is under changing load, you don't have clearly observable conditions. We should therefore avoid these complex cases for the time being as long as there are doubts about MangoHud. 

In the case I described above (steps 1 to 6), the GPU is at 200 Mhz but MangoHud permanently shows 1040 Mhz. Very rarely it goes down to 200 Mhz. This is obviously wrong. Nor can this be explained by average values. But MangoHud nearly always shows 1040 Mhz while Sysfs seems to only return 200 Mhz without a single exception. That simply doesn't make sense. 

I did another test:

```
time while true; do od -An -tu2 -j54 -N14 /sys/class/drm/card0/device/gpu_metrics; sleep 0.01; done > gpu_metrics.log

real	0m44.305s
user	0m6.290s
sys	0m4.919s
```
You can find the resulting file here:
[gpu_metrics.log](https://github.com/ValveSoftware/SteamOS/files/13518356/gpu_metrics.log)

The test ran for just under a minute. MangoHud displayed 200Mhz for three to four seconds, the rest of the time it shows 1040 Mhz. I don't think this is a meaningful representation of the actual metrics. And if that is the case, we don't need to talk any further on the basis of MangoHud and certainly don't need to test more complex scenarios with it. My opinion.

**Comment by Kagukara** [comment]
https://github.com/ValveSoftware/SteamOS/issues/1245#issuecomment-1834681821
> But MangoHud nearly always shows 1040 Mhz

I think that's because MangoHud isn't updating. Are the other numbers in the overlay moving at all (numbers/frametime graph)? As this looks like what I get when I use CSS Loader to stop all animations and the overlay just stops.

**Comment by Oliver-1xok** [comment]
> Can somebody remind me again, what's the actual issue here? That the Steam home screen has animations, so it uses the GPU? Why is that the issue? It's not rendering that when the game is on.
> 

Yes, I ask myself that too. 

> Home/detail screen for each individual game has no animations, so the GPU runs at 200 most of the time, MangoHud numbers freeze because there is no updates to screen (this is as old issue as Steam Deck itself), 

Okay, I didn't know that.  However, the metrics are updated every few seconds on my Deck. Even when the screen has already dimmed. Only the GPU clock speed is almost always set to 1040 Mhz. Only very rarely does it go down to 200 Mhz (the true value). This makes me wonder what is freezing. Freezing would mean that there is a high probability that 200 Mhz would still be displayed. But this is not the case. 

I also suspect that MangoHud jumps up to 1040 far too often in other scenarios. But I can't prove this without putting a lot of effort into it. That's why I wanted to show that MangoHud displays this incorrec value using a simple example. 

At the moment there are many discussions about this oscillation between 200 and 1040 Mhz. However, I have the impression that ManhoHud contributes to this, especially as it don't seems to like values between 200 and 1040 Mhz.

**Comment by Kagukara** [comment]
https://github.com/ValveSoftware/SteamOS/issues/1245#issuecomment-1834796256
> However, I have the impression that ManhoHud contributes to this, especially as it don't seems to like values between 200 and 1040 Mhz.

Does what was said in this comment here not explain why you don't see the values between 200 and 1040MHz?

**Comment by ngoquang2708** [comment]
> > [#1245 (comment)](https://github.com/ValveSoftware/SteamOS/issues/1245#issuecomment-1834796256)
> > However, I have the impression that ManhoHud contributes to this, especially as it don't seems to like values between 200 and 1040 Mhz.
> 
> Does what was said in this [comment](https://github.com/ValveSoftware/SteamOS/issues/1241#issuecomment-1828474301) here not explain why you don't see the values between 200 and 1040MHz?

https://steamcommunity.com/app/1675200/discussions/0/7529517132617601916/?tscn=1701402904#c4038102696018436188

**Comment by ngoquang2708** [comment]
Related PR https://github.com/flightlessmango/MangoHud/pull/1176#issue-2019877828
